moon-store-mac-os/moon-store-mac-os/Services/BaseNetworkService.swift

```diff
@@ -0,0 +1,147 @@
+        urlRequest.httpBody = json
+        urlRequest.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        return try await request(urlRequest: urlRequest)
+    }
+    
+    // MARK: - HTTP POST Multipart
+    
+    func postMultipartData(path: MSPath,
+                           parameters: [String: Any],
+                           dataSet: [Data]) async throws -> Data {
+        components.path = path.endpoint
+        components.queryItems = makeQueryItems(parameters: parameters)
+        
+        guard let url = components.url else {
+            throw NSError(domain: "Invalid url", code: 0)
+        }
+        
+        let boundary = "Boundary-\(UUID().uuidString)"
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "POST"
+        urlRequest.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField: "Content-Type")
+        
+        var data = Data()
+        
+        for (key, value) in parameters {
+            data.appendString("--\(boundary)\r\n")
+            data.appendString("Content-Disposition: form-data; name=\"\(key)\"\r\n\r\n")
+            data.appendString("\(value)\r\n")
+        }
+        
+        for value in dataSet {
+            data.appendString("--\(boundary)\r\n")
+            data.appendString("Content-Disposition: form-data; name=\"images\"; filename=\"image-\(boundary).jpg\"\r\n")
+            data.appendString("Content-Type: image/jpeg\r\n\r\n")
+            data.append(value)
+            data.appendString("\r\n")
+        }
+        
+        data.appendString("--\(boundary)--\r\n")
+        
+        urlRequest.httpBody = data
+        
+        return try await request(urlRequest: urlRequest)
+    }
+    
+    // MARK: - HTTP DELETE
+    
+    func deleteData(path: MSPath, parameters: [String: Any]) async throws -> Data {
+        components.path = path.endpoint
+        components.queryItems = makeQueryItems(parameters: parameters)
+        
+        guard let url = components.url else {
+            throw NSError(domain: "Invalid url", code: 0)
+        }
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "DELETE"
+        
+        return try await request(urlRequest: urlRequest)
+    }
+    
+    // MARK: - HTTP UPDATE
+    
+    func putData(path: MSPath, parameters: [String: Any]) async throws -> Data {
+        components.path = path.endpoint
+        components.queryItems = makeQueryItems(parameters: parameters)
+        
+        guard let url = components.url else {
+            throw NSError(domain: "Invalid url", code: 0)
+        }
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "PUT"
+        
+        return try await request(urlRequest: urlRequest)
+    }
+    
+    func request(urlRequest: URLRequest) async throws -> Data {
+        do {
+            let (data, _) = try await URLSession.shared.data(for: urlRequest)
```

Comment:
Mejor maneja la respuesta (response)
Si no, retorna de un solo request.data

Reply:
Response will be handle in Error handler feature too.